lib/open_food_network/permissions.rb

```diff
@@ -58,55 +58,63 @@ def variant_override_enterprises_per_hub
       permissions
     end
 
-    # Find enterprises that an admin is allowed to add to an order cycle
+    # Find orders that a user can see
     def visible_orders
-      # Any orders that I can edit
-      editable = editable_orders.pluck(:id)
-
-      produced = Spree::Order.with_line_items_variants_and_products_outer.
-        where(
-          distributor_id: granted_distributor_ids,
-          spree_products: { supplier_id: enterprises_with_associated_orders }
-        ).pluck(:id)
-
-      Spree::Order.where(id: editable | produced)
+      Spree::Order.where(id:
+        managed_orders.select(:id) |
```

Comment:
I don't think this avoid extra queries. Our Rails version doesn't have an `or` operator in ActiveRecord. I get this on the console:

```rb
Spree::Order.where(id: Spree::Order.order(:id).limit(3).select(:id) | Spree::Order.order("id desc").limit(3).select(:id))
#  Spree::Order Load (0.9ms)  SELECT id FROM "spree_orders" ORDER BY id LIMIT 3
#  Spree::Order Load (0.7ms)  SELECT id FROM "spree_orders" ORDER BY id desc LIMIT 3
#=>   Spree::Order Load (1.0ms)  SELECT "spree_orders".* FROM "spree_orders" WHERE "spree_orders"."id" IN (1, 2, 4, 577267, 577266, 577265)
```

Reply:
Yes, that's something that's repeated throughout the current permissions logic. The single pipe runs both queries instead of combining them. I think we should be able to make big improvements there with Rails 4...

Reply:
I wonder if we could add a temporary gem to backport some of that functionality until we get to Rails 4?

Reply:
or only comes in rails 5.
I think we should use arel for these cases as suggested in this post.
https://stackoverflow.com/questions/3639656/activerecord-or-query

Reply:
Adding dependencies makes the upgrade harder. The fewer gems, the better.